CMake Error

### Describe the bug

I can  install abacus while using 8d16c36dd86 commit version, but after this, it always throws error about "undefined reference to `rdmft::Veff_rdmft<std::complex<double>, std::complex<double> >::initialize_HR(UnitCell const*, Grid_Driver const*)"

<img width="1698" height="540" alt="Image" src="https://github.com/user-attachments/assets/3442b5d1-a573-4a34-92e5-0e53edd07c22" />

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/source/source_lcao/module_rdmft/rdmft_tools.cpp
+++ b/source/source_lcao/module_rdmft/rdmft_tools.cpp
@@ -186,13 +186,6 @@ double occNum_func(const double eta, const int symbol, const std::string XC_func
     
 }
 
-
-template class Veff_rdmft<double, double>;
-
-template class Veff_rdmft<std::complex<double>, double>;
-
-template class Veff_rdmft<std::complex<double>, std::complex<double>>;
-
 // this part of the code is copying from class Veff
 // initialize_HR()
 template <typename TK, typename TR>
@@ -399,6 +392,10 @@ void Veff_rdmft<double, double>::contributeHR()
 }
 
 }
+template class rdmft::Veff_rdmft<double, double>;
 
+template class rdmft::Veff_rdmft<std::complex<double>, double>;
+
+template class rdmft::Veff_rdmft<std::complex<double>, std::complex<double>>;
 
 

